rules/aip0233/response_message_name.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0233
+
+import (
+	"fmt"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+)
+
+// Batch Create method should have a properly named Response message.
+var responseMessageName = &lint.MethodRule{
+	Name:   lint.NewRuleName(233, "response-message-name"),
+	OnlyIf: isBatchCreateMethod,
+	LintMethod: func(m *desc.MethodDescriptor) []lint.Problem {
+		// Proper response name should be the concatenation of the method name and
+		// "Response"
+		want := m.GetName() + "Response"
+
+		// If this is an LRO, then use the annotated response type instead of
+		// the actual RPC return type.
+		got := m.GetOutputType().GetName()
+		if m.GetOutputType().GetFullyQualifiedName() == "google.longrunning.Operation" {
+			got = utils.GetOperationInfo(m).GetResponseType()
+		}
+
+		// Rule check: Establish that for methods such as `BatchCreateFoos`, the
+		// response message should be named `BatchCreateFoosResponse`
+		//
+		// Note: If `got` is empty string, this is an unannotated LRO.
+		// The AIP-151 rule will whine about that, and this rule should not as it
+		// would be confusing.
+		if got != want && got != "" {
+			return []lint.Problem{{
+				Message: fmt.Sprintf(
+					"Batch Create RPCs should have a properly named response message %q, but not %q",
+					want, got,
+				),
+				Descriptor: m,
```

Comment:
Let's return the precise location of the response type: `locations.MethodResponseType`